### PR TITLE
*: 'load data' should not use latches

### PR DESCRIPTION
**Pull request by tiancaiamao**
## What have you changed? (mandatory)

'load data' is not retryable when it meets conflicts, while latches may result in
false positive transaction conflicts. so enable latches will lead to 'load data'
abort abnormally, even there are no conflicts.

## What are the type of the changes (mandatory)?

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested (mandatory)?

- unit text

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/tidb/pull/6418

@coocood  @AndreMouche @shenli

### CODE REVIEW

File: store/tikv/txn.go [hunk 1]
```diff
@@ -195,12 +195,12 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
...
 
 	// latches enabled
-	var forUpdate bool
-	if option := txn.us.GetOption(kv.ForUpdate); option != nil {
-		forUpdate = option.(bool)
+	var bypassLatch bool
+	if option := txn.us.GetOption(kv.BypassLatch); option != nil {
+		bypassLatch = option.(bool)
 	}
 	// For update transaction is not retryable, commit directly.
```
Comment (coocood) [inline store/tikv/txn.go]: Need to update the comment too.

File: executor/write_test.go [hunk 2]
```diff
@@ -24,12 +24,19 @@ import (
...
 	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/session"
 	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/store/mockstore"
 	"github.com/pingcap/tidb/table"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/testkit"
 )
 
+type testBypassSuite struct{}
```
Comment (coocood) [inline executor/write_test.go]: Why not reuse the `testSuite`?

Comment (tiancaiamao) [reply executor/write_test.go]: This test create a new store to set latch capacity, it's not common in `testSuite`.
And if I write it in `testSuite`, I found `TestShow` fail:

```
FAIL: show_test.go:35: testSuite.TestShow

show_test.go:141:
    result = tk.MustQuery(testSQL)
/media/genius/OS/project/src/github.com/pingcap/tidb/util/testkit/testkit.go:180:
    tk.c.Assert(errors.ErrorStack(err), check.Equals, "", comment)
... obtained string = "" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/store/tikv/snapshot.go:312: tikv aborts txn: leveldb: closed\n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/store/tikv/snapshot.go:314: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/store/tikv/snapshot.go:269: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/store/tikv/snapshot.go:218: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/kv/union_store.go:186: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/store/tikv/txn.go:96: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/structure/list.go:220: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/structure/list.go:164: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/meta/meta.go:499: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/meta/meta.go:512: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/util/admin/admin.go:52: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/ddl/stat.go:58: \n" +
...     "/media/genius/OS/project/src/github.com/pingcap/tidb/kv/txn.go:57: \n" +
```

So I just create a new `Suite`.

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @zhexuany

**Comment by tiancaiamao** [comment]
PTAL @jackysp

**Comment by zhexuany** [comment]
@tiancaiamao Please resolve conflict.

**Pull request merged**